### PR TITLE
[fga] further debug sharing issue

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 328709e</samp>

This pull request refactors and enhances the authorizer module for workspace sharing. It removes redundant code, adds verification and logging, and wraps SpiceDB objects in `TrustedValue` instances.

</details>

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/authorization/authorizer.ts [hunk 1]
```diff
@@ -428,51 +428,6 @@ export class Authorizer {
             rels.push(set(rel.workspace(workspaceID).shared.anyUser));
         }
         await this.authorizer.writeRelationships(...rels);
-
```
Comment (svenefftinge) [inline components/server/src/authorization/authorizer.ts:431]: this was never triggered, so good to remove

File: components/server/src/authorization/spicedb-authorizer.ts [hunk 2]
```diff
@@ -130,7 +130,10 @@ export class SpiceDBAuthorizerImpl implements SpiceDBAuthorizer {
                     this.callOptions,
                 ),
             );
-            log.info("[spicedb] Successfully wrote relationships.", { response, updates });
+            log.info("[spicedb] Successfully wrote relationships.", {
```
Comment (geropl) [inline components/server/src/authorization/spicedb-authorizer.ts:133]: :pray:

**Review by geropl — APPROVED** [review]
LGTM :heavy_check_mark:

**Pull request merged**